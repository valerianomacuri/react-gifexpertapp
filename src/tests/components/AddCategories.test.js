const { shallow } = require("enzyme");
const { AddCategory } = require("../../components/AddCategories");

describe('<AddCategory />', () => {

    const setCategories = jest.fn()
    let wrapper = shallow(<AddCategory setCategories={ setCategories } />)
    beforeEach( () => {
      jest.clearAllMocks()

      wrapper = shallow( <AddCategory setCategories={ setCategories } /> )
    } )
    test('should show correctly', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('should change the text field', () => {
      const input = wrapper.find('input')
        const value = 'Hola Mundo'
      input.simulate('change', {
        target: {
            value
        }
      })

      expect( wrapper.find('input').prop('value') ).toBe( value )
    })

    test('should not post the information with submit action', () => {
      wrapper.find('form').simulate('submit', {
        preventDefault(){}
      })

      expect( setCategories ).not.toHaveBeenCalled()
    });
    
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {

      const value = 'Hola Mundo'

      wrapper.find('input').simulate('change', { target: { value } })

      wrapper.find('form').simulate('submit', { preventDefault(){} })

      expect( setCategories ).toHaveBeenCalledWith(
        // para que el argumento de la función sea una función
        expect.any(Function)
      )

      expect( wrapper.find('input').prop('value') ).toBe('')

    })
});
