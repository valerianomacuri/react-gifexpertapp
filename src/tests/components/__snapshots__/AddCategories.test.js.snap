// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory /> should show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
